In [1]:
import torch
from models import FcEncoder, CnnEncoder, DensenetEncoder, Decoder
from utils import num_of_params

In [2]:
fc_encoder = FcEncoder()
cnn_encoder = CnnEncoder()
densenet_encoder = DensenetEncoder()
print(num_of_params(fc_encoder))
print(num_of_params(cnn_encoder))
print(num_of_params(densenet_encoder))


12845312
5519120
7216256


[Parameter containing:
 tensor([[ 2.9654e-03,  3.7817e-03,  3.1466e-03,  ...,  2.6271e-04,
          -3.1943e-04, -4.3551e-03],
         [-1.2825e-03,  3.4080e-03,  3.5732e-03,  ..., -1.3513e-03,
           2.2347e-03, -2.4321e-03],
         [-3.6132e-03,  3.0265e-04, -4.2604e-03,  ...,  2.0422e-05,
          -3.7495e-03,  5.2963e-04],
         ...,
         [-3.5487e-03, -3.3096e-03,  2.5239e-03,  ...,  3.3567e-03,
           8.3156e-04, -3.0851e-03],
         [-1.7569e-03,  3.4840e-04, -2.1397e-03,  ...,  4.1648e-04,
          -1.8280e-03,  1.5042e-03],
         [-1.2643e-03, -5.8921e-04, -7.7070e-04,  ...,  1.3715e-03,
          -2.1582e-03,  4.2224e-03]], requires_grad=True),
 Parameter containing:
 tensor([ 3.3719e-03, -8.7857e-04, -9.0331e-04, -4.9350e-05, -1.4173e-03,
         -9.6480e-04, -3.4934e-03,  3.3174e-04, -1.8285e-03, -3.1820e-03,
         -4.3264e-03,  2.7693e-03, -2.6025e-03, -1.3299e-03, -2.6506e-03,
         -1.0315e-05,  3.2384e-03,  3.4079e-03,  1.9260e-03, -1.72

In [ ]:
import os
max(os.listdir("data/img_align_celeba/img_align_celeba"))

202599

Images are well-labeled, exactly from 000001 to 202599. No gaps in between

In [13]:
import pandas as pd
labels = pd.read_csv("data/list_attr_celeba.csv", index_col="image_id")

In [15]:
labels.dtypes

5_o_Clock_Shadow       int64
Arched_Eyebrows        int64
Attractive             int64
Bags_Under_Eyes        int64
Bald                   int64
Bangs                  int64
Big_Lips               int64
Big_Nose               int64
Black_Hair             int64
Blond_Hair             int64
Blurry                 int64
Brown_Hair             int64
Bushy_Eyebrows         int64
Chubby                 int64
Double_Chin            int64
Eyeglasses             int64
Goatee                 int64
Gray_Hair              int64
Heavy_Makeup           int64
High_Cheekbones        int64
Male                   int64
Mouth_Slightly_Open    int64
Mustache               int64
Narrow_Eyes            int64
No_Beard               int64
Oval_Face              int64
Pale_Skin              int64
Pointy_Nose            int64
Receding_Hairline      int64
Rosy_Cheeks            int64
Sideburns              int64
Smiling                int64
Straight_Hair          int64
Wavy_Hair              int64
Wearing_Earrin

In [16]:
s = pd.Series([10, 20, 30], index=['a', 'b', 'c'])
s['a']

10